compiler/main/checks.cpp

```diff
@@ -622,6 +624,16 @@ checkResolveRemovedPrims(void) {
   }
 }
 
+static void checkNoRecordDeletes() {
+  // No need to do for_alive_in_Vec - there shouldn't be any, period.
```

Comment:
Being unfamiliar with this code, could you explain this comment a little more? Specifically, what is `for_alive_in_Vec` ? An alternative to `forv_Vec`?